client/verta/tests/unit_tests/test_runtime.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the runtime context logging tools
+"""
+
+from concurrent import futures
+import unittest
+import pytest
+from typing import Any, Dict
+from verta import runtime
+import time
+import random
+
+
+@pytest.fixture(autouse=True)
+def clean_thread():
+    """
+    Ensure thread-local logs are wiped before and after each test.
+    This is located here instead of conftest.py so the use of the
+    autouse flag is limited to this module only.
+    """
+    yield
+    if hasattr(runtime._THREAD, 'logs'):
+        runtime._THREAD.__delattr__('logs')  # clean-up any existing logs.
+
+
+
+def test_thread_safe_context_manager() -> None:
+    """
+    Validate that multiple threads running a context manager simultaneously
+    will stay thread-safe. 50 threads are completed by 10 workers, with
+    each thread verifying that it only holds it own thread-local value for logs.
+    """
+    def log_in_context(value):
+        with runtime.context() as test_ctx:
+            runtime.log('test_key', value)
+            # time.sleep(random.uniform(0, 1))
+        assert hasattr(runtime._THREAD, 'logs') == False
+        assert hasattr(runtime._THREAD, 'validate') == False
+        assert test_ctx.logs() == {'test_key': value}
+
+    with futures.ThreadPoolExecutor(max_workers=10) as executor:
+        list(
+            executor.map(
+                log_in_context,
+                [{f'thread_{x}_key': f'thread_{x}_val'} for x in range(50)]
+            )
+        )
+
+
+def test_init_thread_logs():
+    """
+    An empty dict is initialized by default.
+    """
+    assert runtime._get_thread_logs() == {}
+
+
+def test_set_thread_logs():
+    runtime._set_thread_logs({'test_set': 'test_set'})
+    assert runtime._THREAD.logs ==  {'test_set': 'test_set'}
+
+
+def test_get_thread_logs():
+    runtime._set_thread_logs({'test_get': 'test_get'})
+    assert runtime._get_thread_logs() ==  {'test_get': 'test_get'}
+
+
+def test_init_thread_validate():
+    """
+    A boolean defaulting to False is initialized by default.
+    """
+    assert runtime._get_validate_flag() == False
+
+
+def test_set_thread_validate():
+    runtime._set_validate_flag(True)
+    assert runtime._THREAD.validate == True
+
+
+def test_get_thread_validate():
+    runtime._set_validate_flag(True)
+    assert runtime._get_validate_flag() == True
+    runtime._set_validate_flag(False)
```

Comment:
Why does one test reset


```python
runtime._set_validate_flag(False)
```

at the end but not the other? Maybe it should be in the cleanup fixture? And maybe it should be a `delattr()` instead?